packages/components/pie-modal/test/component/pie-modal.spec.ts

```diff
@@ -25,3 +25,28 @@ Object.values(HEADING_LEVELS).forEach((headingLevel) => test(`should render the
     // h2 is the default / fallback value
     await expect(component.locator(`${HEADING_LEVELS.H2}.c-modal-heading`)).toContainText(props.heading);
 }));
+
+test.describe('`When the Pie Modal is closed`', () => {
+    test('should dispatch event `pie-modal-close`', async ({ mount }) => {
+        const messages: string[] = [];
+        const component = await mount(
+            PieModal,
+            {
+                props: {
+                    isOpen: true,
+                }
+            },
+        );
+
+        await component.update({
+            on: {
+                click: (event: string) => messages.push(event)
+            }
+        });
+
+        await component.locator('.c-modal-closeBtn').click();
+
+        expect(messages).toHaveLength(1);
```

Comment:
I'm not too happy with this part of the test. Have tried to pull out the event name in the expects which I think should be doable but need to read up a bit more to work out the best way to do that. For now it picks up the event length.

Reply:
I'm assuming you've tried something like this?
```suggestion
        expect(messages).toHaveLength(1);
        const [message] = messages;
        expect(message.type).toBe(ON_MODAL_CLOSE_EVENT);
```

Reply:
You could even throw in an `expect(message instanceof CustomError).toBe(true);`

Reply:
I did try the message thing (good shout) but didn't work. I feel there is something else going on so will reattempt this test in my next ticket.